nmos-test.py

```diff
@@ -508,8 +514,19 @@ def validate_args(args):
 
 def start_web_servers():
     port = 5001
+    if ENABLE_HTTPS:
+        ctx = ssl.create_default_context()
+        ctx.options &= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+        for cert, key in zip(CERTS_MOCKS, KEYS_MOCKS):
+            ctx.load_cert_chain(cert, key)
+        # TODO: Provide a mechanism to establish trust for client certificates to avoid check_hostname=False
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
     for app in FLASK_APPS:
-        t = threading.Thread(target=app.run, kwargs={'host': '0.0.0.0', 'port': port, 'threaded': True})
+        if ENABLE_HTTPS:

```

Comment:
Minor thing, but assuming the default argument for ssl_context is None, could we avoid writing the majority of the thread definitions twice and just pass in the ssl_context variable with a different value in each case?

Reply:
I wondered that, and forgot to try it. Will do.